Biswap and Maverick migration

chef-seaweed opened the pull request:
Migrate models/tables in Biswap and Maverick.
Also rename Biswap v1 to v2.

dune-eng commented:
Workflow run id 5797686182 approved.

dune-eng commented:
Workflow run id 5797686324 approved.

dune-eng commented:
Workflow run id 5797775367 approved.

dune-eng commented:
Workflow run id 5797775540 approved.

dune-eng commented:
Workflow run id 5797926746 approved.

dune-eng commented:
Workflow run id 5797927023 approved.

dune-eng commented:
Workflow run id 5798100175 approved.

dune-eng commented:
Workflow run id 5798099827 approved.

dune-eng commented:
Workflow run id 5798214779 approved.

dune-eng commented:
Workflow run id 5798214774 approved.

chef-seaweed commented:
kindly cc @Hosuke @jeff-dude 🙏

dune-eng commented:
Workflow run id 5816253764 approved.

dune-eng commented:
Workflow run id 5816253755 approved.

couralex6 commented on models/maverick/ethereum/maverick_v1_ethereum_trades.sql:
```diff
@@ -21,29 +21,30 @@ WITH dexs AS
+        ,t.contract_address as project_contract_address
         ,t.evt_tx_hash AS tx_hash
-        ,'' AS trace_address
+        ,CAST(NULL as array<bigint>) AS trace_address
```
We started removing `trace_address` from models in dex_trades's upstream lineage, so this isn't needed.

chef-seaweed replied:
Ah sorry missed that one, fix it now.

dune-eng commented:
Workflow run id 5819666970 approved.

dune-eng commented:
Workflow run id 5819666958 approved.

dune-eng commented:
Workflow run id 5822783927 approved.

jeff-dude commented on models/biswap/bnb/biswap_bnb_schema.yml:
```diff
@@ -20,84 +20,57 @@ models:
-          version: 1
+          version: 2
     columns:
-      - &blockchain
```
why remove `&` usage? if you keep, then you can reuse with `*` in following spells added to schema to avoid clutter

chef-seaweed replied:
Sorry didn't know this would cause clutter, fixed now

jeff-dude commented on models/biswap/bnb/biswap_bnb_trades_legacy.sql:
```diff
@@ -1,108 +1,19 @@
-    WHERE t.evt_block_time >= '{{ project_start_date }}'
-    {% endif %}
-)
+{% set biswap_models = [
+    ref('biswap_v2_bnb_trades_legacy')
+,   ref('biswap_v3_bnb_trades_legacy')
+] %}
```
could remove this to keep clean

chef-seaweed replied:
removed now

jeff-dude commented on models/maverick/ethereum/maverick_ethereum_schema.yml:
```diff
@@ -19,78 +19,55 @@ models:
           project: maverick
           version: 1
     columns:
-      - &blockchain
```
same question here

chef-seaweed replied:
fixed now

jeff-dude reviewed (changes requested):
can you look to fix merge conflicts, now that it appears maverick was migrated and merged?

also, minor points below

dune-eng commented:
Workflow run id 5846581838 approved.

dune-eng commented:
Workflow run id 5846589673 approved.

dune-eng commented:
Workflow run id 5846597457 approved.

dune-eng commented:
Workflow run id 5846599734 approved.

dune-eng commented:
Workflow run id 5846599849 approved.

dune-eng commented:
Workflow run id 5846600401 approved.

dune-eng commented:
Workflow run id 5846600434 approved.

dune-eng commented:
Workflow run id 5846697040 approved.

dune-eng commented:
Workflow run id 5846697070 approved.

dune-eng commented:
Workflow run id 5846720758 approved.

dune-eng commented:
Workflow run id 5846720756 approved.

jeff-dude commented:
if you're doing more than just migrating -- such as actual spell changes -- this is where it can be a little more complex.

from what i see, `biswap_bnb_trades` is being repurposed to `biswap_v2_bnb_trades` and then a cross-version spell to bring together `v2` and `v3`. since you're overlapping migration and changes in the same PR, then you will likely need to apply the same changes to both legacy and non-legacy, otherwise `dex.trades` downstream won't have `biswap` data until migration is final.

we shouldn't be overriding existing legacy files to be dummy files, but rather only build out on new dunesql spells. these overrides would actually break `dex.trades` if merged

i think this PR needs all legacy and non-legacy spells of each modified to where they need to be, from each blockchain/version up through the cross-chain spells

dune-eng commented:
Workflow run id 5867904157 approved.

dune-eng commented:
Workflow run id 5868297850 approved.

dune-eng commented:
Workflow run id 5868297840 approved.

dune-eng commented:
Workflow run id 5868859521 approved.

dune-eng commented:
Workflow run id 5868926958 approved.

dune-eng commented:
Workflow run id 5868960098 approved.

dune-eng commented:
Workflow run id 5868959990 approved.

dune-eng commented:
Workflow run id 5868992176 approved.

dune-eng commented:
Workflow run id 5868992190 approved.

chef-seaweed commented:
> if you're doing more than just migrating -- such as actual spell changes -- this is where it can be a little more complex.
> 
> from what i see, `biswap_bnb_trades` is being repurposed to `biswap_v2_bnb_trades` and then a cross-version spell to bring together `v2` and `v3`. since you're overlapping migration and changes in the same PR, then you will likely need to apply the same changes to both legacy and non-legacy, otherwise `dex.trades` downstream won't have `biswap` data until migration is final.
> 
> we shouldn't be overriding existing legacy files to be dummy files, but rather only build out on new dunesql spells. these overrides would actually break `dex.trades` if merged
> 
> i think this PR needs all legacy and non-legacy spells of each modified to where they need to be, from each blockchain/version up through the cross-chain spells

Got it. I've modified `biswap_v2_bnb_trades_legacy` and `biswap_bnb_trades_legacy` to let dex.trades downstream still has `Biswap` data. Also changed `maverick_v1_ethereum_trades_legacy` and `maverick_trades_legacy` back to as-is. 

Please let me know if anything else should be done, thanks @jeff-dude 🙏

dune-eng commented:
Workflow run id 5869128061 approved.

dune-eng commented:
Workflow run id 5869128039 approved.

dune-eng commented:
Workflow run id 5869132358 approved.

dune-eng commented:
Workflow run id 5869132329 approved.
